hedera-node/src/main/java/com/hedera/services/legacy/services/state/AwareProcessLogic.java

```diff
@@ -137,7 +137,7 @@ private void warnOf(Exception e, String context) {
 	void addRecordToStream() {
 		ctx.recordsHistorian().lastCreatedRecord().ifPresent(finalRecord ->
 				addForStreaming(ctx.txnCtx().accessor().getBackwardCompatibleSignedTxn(),
-						finalRecord, ctx.txnCtx().consensusTime()));
+						finalRecord.asGrpc(), ctx.txnCtx().consensusTime()));
```

Comment:
It will be a nice perf win if we can make this `ExpirableTxnRecord.asGprc()` call **in the record stream thread**.

Once option would be to make `RecordStreamObject` accept an `ExpirableTxnRecord`; and then [here](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/stream/RecordStreamObject.java#L87) do instead,
```
		out.writeByteArray(expirableTransactionRecord.asGrpc().toByteArray());
```

Reply:
Agreed. It reduces conversion in other places as well. Made the changes. Thanks Michael ! 🙂